costs/diff/diff_route.go

```diff
@@ -117,7 +118,47 @@ func makeElasticSearchRequest(ctx context.Context, parsedParams esQueryParams) (
 }
 
 // getDiffData returns the cost diff based on the query params, in JSON or CSV format.
-func getDiffData(request *http.Request, a routes.Arguments) (int, interface{}) {
+func getDiffData(ctx context.Context, parsedParams esQueryParams) (int, interface{}) {
+	sr, returnCode, err := makeElasticSearchRequest(ctx, parsedParams)
+	if err != nil {
+		if returnCode == http.StatusOK {
+			return returnCode, nil
+		} else {
+			return returnCode, err
+		}
+	}
+	res, err := prepareDiffData(ctx, sr)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+	return http.StatusOK, res
+}
+
+func convertDiffData(ctx context.Context, diffData interface{}) (costDiff, error) {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	if report, ok := diffData.(costDiff); ok {
+		return report, nil
+	}
+	logger.Error("An error occured while converting to diffData", nil)
+	return nil, fmt.Errorf("Error durint cast")
+}
+
+// TaskDiffData prepares an elasticsearch query and retrieves cost differentiator data
+func TaskDiffData(ctx context.Context, aa aws.AwsAccount) (data costDiff, err error) {
+	now := time.Now()
+	dateBegin := time.Date(now.Year(), now.Month() - 1, 1, 0, 0, 0, 0, now.Location()).UTC()
+	dateEnd := time.Date(now.Year(), now.Month(), 1, 23, 59, 59, 999999999, now.Location()).UTC().AddDate(0, 0, -1)
+	parsedParams := esQueryParams{
+		accountList:       []string{es.GetAccountIdFromRoleArn(aa.RoleArn)},
+		dateBegin: dateBegin,
+		dateEnd: dateEnd,
+		aggregationPeriod: "day",
+	}
+	_, diffData := getDiffData(ctx, parsedParams)
```

Comment:
You don't initialize any index list so the ES request will search on every indexes.